packages/core/src/modules/credentials/protocol/v2/__tests__/v2.ldproof.credentials-auto-accept.test.ts

```diff
@@ -0,0 +1,481 @@
+import type { Agent } from '../../../../../agent/Agent'
+import type { SignCredentialOptions } from '../../../../../modules/vc/models/W3cCredentialServiceOptions'
+import type { ConnectionRecord } from '../../../../connections'
+import type {
+  AcceptOfferOptions,
+  AcceptProposalOptions,
+  NegotiateOfferOptions,
+  NegotiateProposalOptions,
+  OfferCredentialOptions,
+  ProposeCredentialOptions,
+} from '../../../CredentialsModuleOptions'
+
+import { setupCredentialTests, waitForCredentialRecord } from '../../../../../../tests/helpers'
+import testLogger from '../../../../../../tests/logger'
+import { KeyType } from '../../../../../crypto'
+import { Key } from '../../../../../crypto/Key'
+import { AriesFrameworkError } from '../../../../../error/AriesFrameworkError'
+import { DidKey } from '../../../../../modules/dids'
+import { W3cCredential } from '../../../../../modules/vc/models'
+import { JsonTransformer } from '../../../../../utils'
+import { sleep } from '../../../../../utils/sleep'
+import { IndyWallet } from '../../../../../wallet/IndyWallet'
+import { AutoAcceptCredential } from '../../../CredentialAutoAcceptType'
+import { CredentialProtocolVersion } from '../../../CredentialProtocolVersion'
+import { CredentialState } from '../../../CredentialState'
+import { CredentialExchangeRecord } from '../../../repository/CredentialExchangeRecord'
+import { V2CredentialPreview } from '../V2CredentialPreview'
+
+describe('credentials', () => {
+  let faberAgent: Agent
+  let aliceAgent: Agent
+  let credDefId: string
+  let faberConnection: ConnectionRecord
+  let aliceConnection: ConnectionRecord
+  let aliceCredentialRecord: CredentialExchangeRecord
+  let wallet: IndyWallet
+  let issuerDidKey: DidKey
+  let verificationMethod: string
+  let credential: W3cCredential
+  let signCredentialOptions: SignCredentialOptions
+
+  describe('Auto accept on `always`', () => {
+    const seed = 'testseed000000000000000000000001'
+    beforeAll(async () => {
+      ;({ faberAgent, aliceAgent, credDefId, faberConnection, aliceConnection } = await setupCredentialTests(
+        'faber agent: always v2 jsonld',
+        'alice agent: always v2 jsonld',
+        AutoAcceptCredential.Always
+      ))
+      wallet = faberAgent.injectionContainer.resolve(IndyWallet)
+      const issuerDidInfo = await wallet.createDid({ seed })
+      const issuerKey = Key.fromPublicKeyBase58(issuerDidInfo.verkey, KeyType.Ed25519)
+      issuerDidKey = new DidKey(issuerKey)
+      verificationMethod = `${issuerDidKey.did}#${issuerDidKey.key.fingerprint}`
+      const inputDoc = {
```

Comment:
Maybe we can extract this to a `__fixtures__` folder and store them as json. Then we store both the unsigned and the signed variant and we can make sure the output matches the stored fixtures. This makes the tests files smaller and we have all the test data in a single place. We do this in other places in the framework (look for json files / fixtures folders etc..)